### ISSUE TITLE
Implement hash navigation for application state

### ISSUE THREAD

**Issue by DVRPCfreight** [DVRPCfreight/TrafficCountConcept#9]
Require hash location events for various pages of the application.

**Comment by wtsay** [comment]
https://github.com/DVRPCfreight/TrafficCountConcept/commit/bc5a1a88b354cc0e03dd1ee4ea317a45a57c6822

Demo:
http://wololo.co/TCC_i9_bc5a1a8/#pg}xzrFvbuiM|RxCpAwTmS}C
(You'll probably need to copy pasta this link. Github transforms the braces to their % equivalent)

My interpretation of the glorious leader's intent. And it seems useful to be able to share precise search windows.

**Comment by wtsay** [comment]
Definitely created a bug - Search by Municipality borkage
Search by municipality uses L.GeoJSON. L.Polygon had methods injected into it via Polyline.encoded.js

**Comment by wtsay** [comment]
Demo:
http://wololo.co/TCC_i9_d12bf87/#pgoqwrFrosiMxAR%5EPqA%40aGDwDByBYeAO%7BDg%40YtEIxAEb%40CXc%40rGGbAObCStDQzCSdDi%40tHe%40zH_%40jGEj%40%7BBY%7DC_%40_BSs%40KeBUc%40EmAMsBYgC%5DMCs%40KcBU%7DASUCWCm%40GwB%5BoEm%40YEkDc%40aBUuAS%7BDg%40cBS%7BAQc%40E%5DE%5DGYEIAIASCMCOCUEKCMCIAICGAEAEAWCc%40Ge%40GaAMKASAR%7DCR%7DC%40O%40M%40M%40Q%3FO%3FS%3FSAO%3FU%3FQ%40Y%40Y%40SHsAL_BF%7B%40PoC%40O%40SJ%7BAV_EDs%40B%5BLgBLaCD_AHiAPmC%40MJ_BB%5DF%7B%40FaAHkAFy%40%40OBW%40IJiBB_%40B_%40FgA%3FCF%7D%40%40QB%5D%40I%40IJFZJn%40P%60%40HNDvCr%40XFRa%40h%40eAVyGPeEn%40aPjHnBNDlBd%40%60E~%40fYzEbGz%40zc%40RyDfRUdAElCq%40bIlAr%40ZVkDATTn%40zAb%40KhAWIb%40ETI%5EEJ%40PATb%40RlALXBb%40Fp%40Hb%40DxA%40

Look at that. Man. Encoded URL for 19106

**Comment by crvanpollard** [comment]
Is Koop something worth looking into? http://koopjs.github.io 
Possibly use for storing those embedded URL into a database for a certain time duration

**Comment by wtsay** [comment]
Based on the front page synopsis, I would be hesitant to implement it. It apparently requires Node.js (server side "issues" getting that working) and is itself a web server.

If the main concern is URL shortening, it seems simple enough to write a server side script that does the DB management and redirection. 

IF I were to implement this independently:
Creating a shortened link
1. Button tied to a PHP script that GET/POSTs whatever hash/URI data
2. PHP generates a unique string, throws that and the URI data into MySQL
3. PHP returns the unique string

Accessing the shortened link
1. PHP script that reads the unique string in the URL
2. Queries MySQL, grabs the relevant URI and formats that into a URL
3. Redirect to the formatted link.

Basically, it would need some sort of DB and server side language. (Neither of which are available, to my knowledge.)

**Comment by wtsay** [comment]
This is like, 90% of the way there.

Municipal bounded search:
http://wololo.co/TCC_i9_f4516fa/#mun_boundary/3188

Record number search:
http://wololo.co/TCC_i9_f4516fa/#rec/1%2C2%2C3%2C4%2C5%2C6%2C7%2C8%2C10

Zip code search:
http://wololo.co/TCC_i9_f4516fa/#zip/5380

Polygon search:
http://wololo.co/TCC_i9_f4516fa/#poly/ik%60uFbdsiMp%7B%40%60s%40rk%40oc%40gBkkBgv%40re%40lc%40qpAyu%40uPsm%40ePuBn~Ao%5EmTwF%60qAzSdPhVwg%40Xtf%40

**Comment by DVRPCfreight** [comment]
@crvanpollard What you were asking for has been mostly worked out by the unmatched skills of @wtsay 

We will need to make some adjustments but the logic is all there

**Comment by crvanpollard** [comment]
@DVRPCfreight @wtsay Great, now how do we get it to work with the current version of TC web app...cuz I don't know where wololo lives?
